### PR TITLE
add waittime-parameter

### PR DESCRIPTION
**Pull request by Thomas--F**
please check and test the code for issue  #750

It works fine for me, but I only tested the mysql-part. The other daos are changed too and I think they're working, but I cannot be sure.

The new parameter "Waittime" at the sources skips a feet-update, if the amount of seconds in this field isn't reached after the last update.
You can still update the source manually from the sources-drop-down.

### CODE REVIEW

File: helpers/ContentLoader.php [hunk 1]
```diff
@@ -43,7 +43,13 @@ public function __construct() {
     public function update() {
         $sourcesDao = new \daos\Sources();
         foreach($sourcesDao->getByLastUpdate() as $source) {
-            $this->fetch($source);
+        	  $last = $source['lastupdate'];
+        	  $wait = $source['waitperiod'];
+        	  if (($last + $wait) < time()){
+	            $this->fetch($source);
+//            } else {
```
Comment (jtojnar) [inline helpers/ContentLoader.php]: Why is this commented code here?

File: daos/pgsql/Database.php [hunk 2]
```diff
@@ -199,6 +199,13 @@ public function __construct() {
                         INSERT INTO version (version) VALUES (9);
                     ');
 				}
+                if(strnatcmp($version, "11") < 0) {
+                    \F3::get('db')->exec(array(
+                        'ALTER TABLE `' . \F3::get('db_prefix') . 'sources` ADD waitperiod INT NOT NULL DEFAULT 0;'
```
Comment (jtojnar) [inline daos/pgsql/Database.php]: Missing comma after first array item.

File: templates/source.phtml [hunk 3]
```diff
@@ -57,6 +57,9 @@
 
         <!-- filter -->
         <li><label for="filter-<?= $idAttr ?>"><?PHP echo \F3::get('lang_source_filter')?>:</label> <input id="filter-<?= $idAttr ?>" type="text" name="filter" value="<?PHP echo isset($this->source) ? $this->source['filter'] : '' ; ?>" placeholder="<?PHP echo F3::get('lang_source_optional'); ?>" /></li>
+
+        <!-- waitperiod -->
+        <li><label for="filter-<?= $idAttr ?>"><?PHP echo \F3::get('lang_source_waitperiod')?>:</label> <input id="filter-<?= $idAttr ?>" type="text" name="waitperiod" value="<?PHP echo isset($this->source) ? $this->source['waitperiod'] : '' ; ?>" placeholder="<?PHP echo F3::get('lang_source_seconds'); ?>" /></li>
```
Comment (Thomas--F) [inline templates/source.phtml]: good point. done

File: helpers/ContentLoader.php (hunk 1 above)
Comment (Thomas--F) [inline helpers/ContentLoader.php]: If you activate these lines you get a feedback on the cliupdate when a feed is skipped. I wasn't sure if that's ok.

**Comment by Thomas--F** [comment]
@jtojnar: Thanks for yout help.

About the commented lines... should I use them or delete them?

File: helpers/ContentLoader.php (hunk 1 above)
Comment (jtojnar) [inline helpers/ContentLoader.php]: It would make sense to log it, I am not sure where should the echoed output appear.

**Comment by jtojnar** [comment]
I would replace the commented line with proper logging.

Additionally, it might be nice to show the last updated time or that it was skipped on the source management and/or source view page.

**Comment by Thomas--F** [comment]
I changed the output to the logger (log-level INFO). The local test was fine.

**Comment by jtojnar** [comment]
I tested it with other database engines. SQLite seems to work fine but PostgreSQL fails when upgrading the database to version 11. It turns out the backticks used for enclosing identifiers are [non-standard](https://stackoverflow.com/questions/10573922/what-does-the-sql-standard-say-about-usage-of-backtick) and SQLite only recognizes them [to be more compatible with MySQL](https://sqlite.org/lang_keywords.html). Double quotes should be used instead, afterwards it is good to go.

**Comment by Thomas--F** [comment]
I removed the backticks and the db_prefix, because the other statements dont't use them too.

**Comment by jtojnar** [comment]
Yes, that is another problem, though people who run PostgreSQL databases probably won’t need such a last-resort measures like table prefixes; that is intended mainly for extremely low cost shared hostings providing only one database.

One last thing I noticed, the English “Waittime” label should probably be two words.

**Comment by Thomas--F** [comment]
Fixed it.

**Comment by SSilence** [comment]
Hm I don't think this is a really necessary usecase. Traffic is a very cheap thing and I think no provider has a problem if my selfoss pulls a rss feed every 30 minutes which updates rarely.

**Comment by Thomas--F** [comment]
It's ok. Your software - your rules ;-)

For me this is a very useful feature becaus I run several services on my RaspberryPie and this little things tends to slow down when receiving many feeds at the same time.
I will try to stay up-to-date with my fork whenever you release a new version, so everyone who wants this feature can download it there.

Keep up the good work! Thank you for this very useful software!

**Comment by jtojnar** [comment]
@Thomas--F https://github.com/SSilence/selfoss/wiki/Maintained-feature-patches

**Comment by Thomas--F** [comment]
I updated my fork to the current version (some commits after release 2.17)

**Pull request closed without merging**